### PR TITLE
Multiverse GC improvements + permanent store flush fix

### PR DESCRIPTION
**Pull request by eugene-babichenko**
- Make `Multiverse::gc` available for any `State`.
- Allow to configure it with the desired depth (to be used with epoch stability depth).

### CODE REVIEW

**Review by mzabaluev — APPROVED** [review]

**Review by mzabaluev — APPROVED** [review]
LGTM. The lint failures have been introduced to chain-vote that's already in master, #454 should fix at least some.

File: chain-storage/src/value.rs [hunk 1]
```diff
@@ -72,3 +72,13 @@ impl From<Vec<u8>> for Value {
         Self::owned(value.into_boxed_slice())
     }
 }
+
+impl fmt::Display for Value {
```
Comment (mzabaluev) [inline chain-storage/src/value.rs:76]: I'd rather have a custom `Debug` rather than use `Display` for this.
A `Value` could be as large as a block, right?

Comment (eugene-babichenko) [reply chain-storage/src/value.rs:79]: Maybe. The idea was to use it display hex block ids.

Comment (mzabaluev) [reply chain-storage/src/value.rs:79]: Perhaps it's better to dump specific type instances after parsing?

Comment (eugene-babichenko) [reply chain-storage/src/value.rs:79]: Yeah probably you're right.

**Pull request merged**